Shared/IISFunctions/Get-IISModules.ps1

```diff
@@ -0,0 +1,159 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+function Get-IISModules {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [System.Xml.XmlNode]$ApplicationHostConfig,
+
+        [Parameter(Mandatory = $false)]
+        [scriptblock]$CatchActionFunction
+    )
+
+    begin {
+        Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+        function ConvertCmdEnvironmentVariableToFullPath {
+            [CmdletBinding()]
+            [OutputType([System.String])]
+            param(
+                [String]$PathWithVar
+            )
+
+            $returnPath = $null
+            if (-not([String]::IsNullOrEmpty($PathWithVar))) {
+                # Assuming that we have the env var always at the beginning of the string and no other vars within the string
+                # Example: %windir%\system32\someexample.dll
+                $preparedPath = ($PathWithVar.Split("%", [System.StringSplitOptions]::RemoveEmptyEntries))
+                if ($preparedPath.Count -eq 2) {
+                    if ($preparedPath[0] -notmatch "\\.+\\") {
+                        $varPath = [System.Environment]::GetEnvironmentVariable($preparedPath[0])
+                        $returnPath = [String]::Join("", $varPath, $($preparedPath[1]))
+                    }
+                }
+            }
+            return $returnPath
+        }
+
+        function GetIISModulesLoaded {
+            [CmdletBinding()]
+            param(
+                [Parameter(Mandatory = $true)]
+                [System.Xml.XmlNode]$Xml
```

Comment:
Should change to the section that we want to pass or have the variable name be `$ApplicationHostConfig`

Reply:
Done, changed function to `GetIISModulesSignatureStatus` and we're now passing `[object[]]$Modules` to it.